clang/lib/Sema/SemaDecl.cpp

```diff
@@ -19001,9 +19001,9 @@ void Sema::ActOnFields(Scope *S, SourceLocation RecLoc, Decl *EnclosingDecl,
 
   // Verify that all the fields are okay.
   SmallVector<FieldDecl*, 32> RecFields;
-
+  std::optional<const FieldDecl *> PreviousField;
```

Comment:
It doesn't seem idiomatic to use an optional pointer, when nullable pointers are built into the language. Consider helping to reduce our collective carbon footprint by saving an extra template instantiation today, and using nullptr instead. =D